pandas/core/arrays/string_arrow.py

```diff
@@ -273,9 +275,22 @@ def __len__(self) -> int:
         """
         return len(self._data)
 
-    @classmethod
-    def _from_factorized(cls, values, original):
-        return cls._from_sequence(values)
+    @doc(ExtensionArray.factorize)
+    def factorize(self, na_sentinel: int = -1) -> Tuple[np.ndarray, ExtensionArray]:
+        encoded = self._data.dictionary_encode()
+        indices = pa.chunked_array(
+            [c.indices for c in encoded.chunks], type=encoded.type.index_type
+        ).to_pandas()
+        if indices.dtype.kind == "f":
+            indices[np.isnan(indices)] = na_sentinel
+        indices = indices.astype(np.int64, copy=False)
```

Comment:
Wondering, is the `int64` needed here? (pyarrow will typically use int32 as default I think) 

I suppose that we always return `int64` from `factorize` for the indices. Short-term, casting to int64 might be best then (to ensure nothing else breaks because of not doing that), but long term we should maybe check if internally we require int64 or would be fine with int32 as well.

Reply:
> Wondering, is the `int64` needed here? (pyarrow will typically use int32 as default I think)

refactor in https://github.com/pandas-dev/pandas/pull/38007/commits/0023f0843cb7fe316a5b30393fef8957b5213d21 partially to address comments

but yes, we seem to be getting an int32 from pyarrow

also we could maybe work with numpy arrays here directly for the indices instead of pandas Series?